Add consul-k8s cli debug command docs (#1308)
hashicorp/web-unified-docs: content/consul/v1.22.x/content/docs/reference/cli/consul-k8s.mdx

  - [`config`](#config): Interact with helm configuration.
      - [`config read`](#config-read): Read helm configuration of a Consul installation.

>   - [`debug`](#debug): Collect configs, logs, and other data from a Consul on Kubernetes deployment in a archive/dir.  

Reviewer: ```suggestion
  - [`debug`](#debug): Collect configs, logs, and other data from a Consul on Kubernetes deployment in an archive directory.  
```

  name: consul
```

### `debug`

> The debug command collect configs, logs, and other data from a Consul on Kubernetes deployment in a archive/dir.

Reviewer: ```suggestion
The debug command collect configs, logs, and other data from a Consul on Kubernetes deployment into an archive directory.
```

```shell-session
$ consul-k8s debug <OPTIONS>
```

The following Command options are available.

| Flag                            | Description                                                                                                                                                                                                                                                                                                                                                                                       | Default                                                     |
| ------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------|

> | <nobr>`-duration`, `-d`</nobr>  | `time.Duration` Sets the time duration to capture logs of consul cluster. (e.g., '5m' for the logs of next 5 minutes). <br/> Minimum time duration should be atleast 10s.                                                                                                                                                                                                                         | `5m0s`                                                      |

Reviewer: ```suggestion
| <nobr>`-duration`, `-d`</nobr>  | `time.Duration` Sets the time duration to capture logs of the Consul cluster. For example, '5m' for the logs of next 5 minutes. <br/> Minimum time duration should be at least 10s.                                                                                                                                                                                                                         | `5m0s`                                                      |
```

> | <nobr>`-since`, `-s`</nobr>     | `time.Duration` Sets the time duration from which to start capturing logs from consul cluster. (e.g., '5m' for the logs of last 5 minutes). <br/> Minimum time duration should be atleast 10s.                                                                                                                                                                                                    | none                                                        |

Reviewer: ```suggestion
| <nobr>`-since`, `-s`</nobr>     | `time.Duration` Sets the time duration from which to start capturing logs from the Consul cluster. For example, '5m' for the logs of last 5 minutes. <br/> Minimum time duration should be at least 10s.                                                                                                                                                                                                    | none                                                        |
```

The following Global options are available.

| Flag                          | Description                                                                                                  | Default                                 |
| ----------------------------- | ------------------------------------------------------------------------------------------------------------ | --------------------------------------- |
| <nobr>`-kubeconfig`</nobr>    | `String` Sets the Kubernetes context to use for Consul K8s CLI operations.                                   | none                                    |
| <nobr>`-kubecontext`</nobr>   | `String` Specifies the path to the kubeconfig file.                                                          | none                                    |
Note: Aliases are not available for the `debug` command global options.

#### Capture Targets

> The `-capture` flag can be specified multiple times to capture specific components before running debug command.

Author: updated it.

> Supported values are: `all`, `helm`, `crds`, `sidecar`, `pods`, `proxy`. By default, it captures `all` components.

Reviewer: ```suggestion
Supported values are: `all`, `helm`, `crds`, `sidecar`, `pods`, `proxy`. By default, the `-capture` flag captures `all` components.
```

| Capture Targets         | Description                                                                                                                                                                                                                                 |
| ----------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | 
| <nobr>`helm`</nobr>     | Captures the Helm release, which is an instance of a Helm chart running in your Kubernetes cluster.                                                                                                                                         | 

> | <nobr>`crds`</nobr>     | Captures both HashiCorp-specific and external CRDs installed by Consul, and list the applied resources for each CRD (e.g., servicerouters, apigateways). <br/> If multiple errors occured, they are captured in a single error file.        | 

Author: replacing "If multiple errors occured, they are captured in a single error file." 
with "If multiple errors occur during capture, this target writes them to a single error file."

So while capturng, if multiple errors occur, then this target writes them to file, otherwise it prints error over terminal. i believe the if statement makes more sense.

| <nobr>`sidecar`</nobr>  | Captures list of all pods across namespaces that have been injected with Consul sidecars using the label `consul.hashicorp.com/connect-inject-status=injected`.                                                                               | 

> | <nobr>`pods`</nobr>     | Captures logs from key Consul component pods: consul-server, consul-client, connect-injector & gateways. <br/> Captures 'log capture status' in an audit file. <br/> If multiple errors occured, they are captured in a single error file.      |

Reviewer: ```suggestion
| <nobr>`pods`</nobr>     | Captures logs from key Consul component pods: consul-server, consul-client, connect-injector, and gateways. <br/> Captures 'log capture status' in an audit file. <br/> This target captures multiple errors in a single error file.      |
```

| <nobr>`proxy`</nobr>    | Captures envoy admin endpoint data (/stats, /clusters, /listeners, /config_dump) for all proxies within cluster. <br/> If multiple errors occured, they are captured in a single error file.                                                | 
| <nobr>`all`</nobr>      | Captures all of the above targets.                                                                                                                                                                                                          | 

#### Sample Debug command

> Below command will capture everything for 10s and write the debug bundle in `debugfile`.

Reviewer: ```suggestion
The following example captures everything for 10s and writes the debug bundle in `debugfile`.
```

Please Note: `-output` `debugfile` is not a complete path, so command will write all data in current working dir `cwd/debugfile`.

```shell-session
$ consul-k8s debug -output debugfile -d 10s archive=false

Starting debugger: 
 - Output:           debugfile
 - Capture Targets:  helm, crds, logs, sidecar, proxy
 ✓ Helm capture successful
 ✓ Crds capture successful
 ✓ Sidecar capture successful
 ✓ Proxy capture successful

Capturing pods logs.....
 - Total Pods:        1x
 - Total Containers:  2x
 - Duration:          10s
 ✓ Logs capture successful
 ✓ Index capture successful
Saved debug directory: debugfile
```

> #### Understanding debug Archive/Directory structure

Author: updated it.

Since the `debug` command collects data from across the cluster, it can be very big archive. So, we've organized the debug bundle and its data using the following patterns 
to help users navigate, read, and share different files more effectively.

> Please note that file structure pattern are only for `logs` and `proxy` target capture as they interact will multiple pods in a cluster.

Reviewer: ```suggestion
Note that file structure patterns are only for `logs` and `proxy` target capture as they interact with multiple pods in a cluster.
```

Author: updated it.

> | Capture Targets         | File structure patterns and naming conventions.                                                                                                                                                                                                                     |

Reviewer: ```suggestion
| Capture targets         | File structure patterns and naming conventions.                                                                                                                                                                                                                     |
```

| ----------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| <nobr>`helm`</nobr>     | `debugfile/helm-config.json`                                                                                                                                                                                                                                        |
| <nobr>`crds`</nobr>     | `debugfile/CRDsResources.json` <br/> `debugfile/CRDsResourcesErrors.txt`                                                                                                                                                                                            |
| <nobr>`sidecar`</nobr>  | `debugfile/sidecar-pods.json`                                                                                                                                                                                                                                       |

> | <nobr>`pods`</nobr>     | pattern: `debugfile/logs/<componentType>/<componentName>/<podName>/<containerName.log>`, <br/> e.g., `debugfile/logs/statefulsets/consul-server/consul-server-0/consul.log` <br/><br/>`debugfile/logCaptureAudit.txt` <br/> `debugfile/logCaptureErrors.txt`        |

Reviewer: ```suggestion
| <nobr>`pods`</nobr>     | pattern: `debugfile/logs/<componentType>/<componentName>/<podName>/<containerName.log>`, <br/> example: `debugfile/logs/statefulsets/consul-server/consul-server-0/consul.log` <br/><br/>`debugfile/logCaptureAudit.txt` <br/> `debugfile/logCaptureErrors.txt`        |
```

> | <nobr>`proxy`</nobr>    | pattern: `debugfile/proxy/<namespace>/<proxyType>/<proxyPod>/*`, <br/> e.g., `debugfile/proxy/consul-ns/sidecar/app-sidecar/config_dumps.json` <br/> `debugfile/proxy/consul-ns/sidecar/app-sidecar/config.json`<br/> `debugfile/proxy/consul-ns/sidecar/app-sidecar/stats.json` <br/><br/>`debugfile/proxyList.json` <br/>`debugfile/proxy/proxyCaptureErrors.txt`                                                                                                                 |

Reviewer: ```suggestion
| <nobr>`proxy`</nobr>    | pattern: `debugfile/proxy/<namespace>/<proxyType>/<proxyPod>/*`, <br/> example: `debugfile/proxy/consul-ns/sidecar/app-sidecar/config_dumps.json` <br/> `debugfile/proxy/consul-ns/sidecar/app-sidecar/config.json`<br/> `debugfile/proxy/consul-ns/sidecar/app-sidecar/stats.json` <br/><br/>`debugfile/proxyList.json` <br/>`debugfile/proxy/proxyCaptureErrors.txt`                                                                                                                 |
```

│   │           └── my-app.log
│   └── statefulsets
│       └── consul-server
│           └── consul-server-0
│               ├── consul.log
│               └── locality-init.log
├── proxy
│   ├── consul
│   │   └── API Gateway
│   │       └── hell-gateway-b85b6644f-6wbcc
│   │           ├── config_dumps.json
│   │           ├── config.json
│   │           └── stats.json
│   ├── default
│   │   └── Sidecar
│   │       └── my-app-7bc65d4d49-smv2s
│   │           ├── config_dumps.json
│   │           ├── config.json
│   │           └── stats.json
│   └── proxyList.json
└── sidecar-pods.json
```

#### Signal Interrupt

<Note>
If the debug run is interrupted, all collected data —including any partial captures, will be cleaned up. 
Please re-run the command to collect all data, or use the `-capture` flag to specify which data to collect.
</Note>

```shell-session
$ consul-k8s debug -output debugfile -d 10s archive=false

Starting debugger: 
  .....
 ✓ Sidecar capture successful
^C ! 
Debug run interrupted (received signal interrupt)
 - Cleaning up partial capture...
 - Cleanup completed
 --> Please re-run the debug command for complete capture
```

> #### Example Commands

Reviewer: ```suggestion
#### Example commands
```

> Below command collects every component data and logs for next 5 miunutes and writes the debug archive in current working directory with foldername as consul-k8s-debug-currentTimestamp. 

Reviewer: ```suggestion
The `consul-k8s debug` command collects all component data and logs for the next five minutes and writes the debug archive in the current working directory with a folder name of `consul-k8s-debug-` plus the current timestamp. 
```

Author: updated it.

```shell-session
$ consul-k8s debug
```

> Below command collects every component data and logs for next 10 minutes and writes the debug archive in current working directory with foldername as consul-k8s-debug-currentTimestamp.

Reviewer: ```suggestion
This example collects all component data and logs for the next ten minutes and writes the debug archive in the current working directory with a folder name of `consul-k8s-debug-` plus the current timestamp. 
```

```shell-session
$ consul-k8s debug -duration 10m
```

> Below command collects every component data and logs of past 10 minutes and writes the debug archive in current working directory with foldername as consul-k8s-debug-currentTimestamp.

Reviewer: ```suggestion
This example collects all component data and logs of the past ten minutes and writes the debug archive in the current working directory with a folder name of `consul-k8s-debug-` plus the current timestamp. 
```

```shell-session
$ consul-k8s debug -since 10m 
```

Below command collects only helm config and logs for next 5 minutes and writes the debug archive in current working directory with foldername as consul-k8s-debug-currentTimestamp.
```shell-session
$ consul-k8s debug -capture helm, logs 
```

> Below command collects only helm config and logs for next 5 minutes and and writes the debug archive in current working directory with foldername as debugfile

Author: updated it.
